### ISSUE TITLE
Put aria-deprecated-role (from aria-roles) into its own rule

### ISSUE THREAD

**Issue by WilcoFiers** [dequelabs/axe-core#4069]
Part of https://github.com/dequelabs/axe-core/issues/4066

The `aria-deprecated-role` check should run in its own rule, instead of as part of `aria-roles`.

**Comment by padmavemulapati** [comment]
Validated with the latest develop branch code base, and seeing that 
`aria-deprecated-role` rule is running as its own rule not with `aria-roles`




![Image](https://github.com/dequelabs/axe-core/assets/47409579/d23a6af6-dd57-43eb-8e69-f6689f94a10a)

with the old version (4.7.0) failures can see under `aria-roles`



![Image](https://github.com/dequelabs/axe-core/assets/47409579/429e870f-e2e9-4b84-8a95-ca70062a8413)
